R/getPrevalence.R

```diff
@@ -580,19 +584,46 @@ setMethod("agglomerateByPrevalence", signature = c(x = "SummarizedExperiment"),
             if(!is.null(rank)){
                 rowData(other_x)[,rank] <- other_label
             }
-            # temporary fix until TSE supports rbind
-            class <- c("SingleCellExperiment","RangedSummarizedExperiment",
-                       "SummarizedExperiment")
-            class_x <- class(x)
-            if(!(class_x %in% class)){
-                class <- "SingleCellExperiment"
-            } else {
-                class <- class[class_x == class]
-            }
-            x <- rbind(as(x[f,],class),
-                       as(other_x,class))
-            x <- as(x,class_x)
+            x <- rbind(x[f,], other_x)
         }
         x
     }
 )
+
+#' @rdname agglomerate-methods
+#' @export
+setMethod("agglomerateByPrevalence", 
+          signature = c(x = "TreeSummarizedExperiment"),
+    function(x, rank = NULL, other_label = "Other", agglomerate.tree = FALSE, 
+            ...){
+      # input check
+      if(!.is_a_bool(agglomerate.tree)){
+        stop("'mergeTree' must be TRUE or FALSE.", call. = FALSE)
+      }
+      # for optionally merging referenceSeq
+      f <- rownames(x) 
+      refSeq <- NULL
+      # mergeRefSeq is a hidden parameter as for all other agglomeration methods
+      # from the agglomerate-methods man page.
+      # Here 'list(...)[["mergeRefSeq]]' is used to access it.
+      mergeRefSeq <- list(...)[["mergeRefSeq"]]
+      if( "mergeRefSeq" %in% names(list(...)) && 
+          !(is.null(mergeRefSeq) || 
+            .is_a_bool(mergeRefSeq)) ){
```

Comment:
It is enough to test

```
merge_refseq <- list(...)[["update.refseq"]]
!is.null(merge_refseq) && !.is_a_bool(merge_refseq)
```

If that parameter is not available the value is NULL. If the value is not NULL and not boolean, it should lead to error.

Reply:
Let's get back to drawing board, this is too complex.

```
merge_refseq <- list(...)[["update.refseq"]]
if( is.null(merge_refseq) {
merge_refseq <- FALSE
}
if( !.is_a_bool(merge_refseq) ){
error
}
if( merge_refseq ){
refSeq <- referenceSeq(x)
}
....
...
...

if( merge_refseq ){
referenceSeq(x) <- .merge_refseq_list(refSeq, f, rownames(x), ...)
}
```